agent/tool-scripts/postprocess/perf-postprocess

```diff
@@ -22,30 +23,49 @@ my $total_samples = 0;
 
 my $line;
 
-open(TXT, "$dir/$file") || die "could not find $dir/$file\n";
-while (my $line = <TXT>) {
-	chomp $line;
-	#    32.95%        928412         qemu-kvm  [kernel.kallsyms]             [k] _raw_spin_lock
-	#     5.50%        242887         qemu-kvm  [kernel.kallsyms]             [k] kvm_vcpu_on_spin
-	#     3.66%        261976         qemu-kvm  [kernel.kallsyms]             [k] update_cfs_rq_blocked_load
-
-	if ($line =~ /^\s*(\d+\.\d+)%\s+(\d+)\s+(\S+)\s+\[(\S+)\]\s+\[(\S+)\]\s+(\S+)/) {
-		$pct = $1;
-		$samples = $2;
-		$bin = $3;
-		$object = $4;
-		$mode = $5;
-		$symbol = $6;
-		# print "found pct: [$pct] samples: [$samples] bin: [$bin] object: [$object] mode: [$mode] symbol [$symbol]\n";
-		$symbol_samples{$symbol} += $samples;
-		$total_samples += $samples;
+if ($callgraph)
+{
+	my $flamegraph_pl=`rpm -qa flamegraph`;
+	my $stackcollapse_perf_pl=`rpm -qa flamegraph-stackcollapse-perf`;
+
+	if ($flamegraph_pl && $stackcollapse_perf_pl){
+		system "perf script -i ${dir}/perf.data > ${dir}/out.perf";
+		system "stackcollapse-perf.pl ${dir}/out.perf > ${dir}/out.folded";
+		system "flamegraph.pl ${dir}/out.folded > ${dir}/flamegraph.svg";
```

Comment:
I wonder if both `stackcollapse-perf` and `flamegraph` can be invoked directly via Perl instead of `system` statements?  Can you research that briefly?

Reply:
@portante  @sourabhtk37 @ndokos  I looked for some potential solutions [here](https://bioinformaticsreview.com/20180506/how-to-execute-unix-shell-commands-in-a-perl-script/) and I found 5 ways to achieve the same 1)`exec` 2)`system` 3)`Backticks “ or qx//` 4)`IPC::Open2` 5)`IPC::Open3`. For simplicity I have used blackticks for now. It is a Perl operator (perlop) very similar to system(), execute a command and then resumes the Perl script after the execution has finished but the return value is STDOUT of the command.

Reply:
I looked and did not see a Perl module for either of those operations, so using back ticks is fine.